contentcuration/contentcuration/serializers.py

```diff
@@ -557,16 +558,38 @@ def retrieve_metadata(self, node):
                 "has_changed_descendant": node.changed,
             }
 
+    def retrieve_source_channel(self, node):
```

Comment:
Note: This method now actually returns the original channel, but there's a method below already called `retrieve_original_channel`, however it seems to be returning something different (it seems to be `None` for a lot of cases where the `Channel.objects.get(pk= node.original_channel_id)` implementation would return something, from brief testing.

So, I haven't spent the time yet to look into how the implementation of `retrieve_original_channel()` works and its use cases. Happy to look into this or if you have thoughts please let me know @jayoshih  —  I'd love to consolidate or otherwise provide a more meaningful name than the currently misleading `retrieve_source_channel` name.

Reply:
If you'd like, you can update that part of the code to use this new method of using the `original_channel_id` field and return the channel along with the thumbnail ([code is here](https://github.com/learningequality/studio/blob/develop/contentcuration/contentcuration/serializers.py#L560)). This is used to display where content was imported from originally here:
![image](https://user-images.githubusercontent.com/7447496/41684384-34361d64-7492-11e8-88e3-7b768b1b743d.png)

Reply:
Awesome! This is really helpful. Changed as suggested, and tested that the old use case of `retrieve_original_channel` "Imported from:" still works (import something from another channel, open it, see the "Imported from:" is correct).